### PR TITLE
new: returns `Issue` from anything that derives `std::error::Error`

### PR DESCRIPTION
**Pull request by KennedyTedesco**
- Returns an `Issue` from anything that derives `std::error::Error`
 - Added new method `Issue::from_string()`

### CODE REVIEW

File: src/issue.rs [hunk 1]
```diff
@@ -187,6 +192,36 @@ impl Issue {
...
+/// use ara_reporting::issue::Issue;
+/// use ara_reporting::issue::IssueSeverity;
+///
+/// let issue = Issue::from("some error just happened");
+/// assert_eq!(IssueSeverity::Error, issue.severity);
+/// assert_eq!("some error just happened", issue.message);
+/// assert_eq!(None, issue.code);
+///
+/// let error: std::num::ParseIntError = "NaN".parse::<u8>().unwrap_err();
+/// let issue: Issue = Issue::from(error.to_string());
```
Comment (azjezz) [inline src/issue.rs]:
```suggestion
/// let error: std::num::ParseIntError = "NaN".parse::<u8>().unwrap_err();
/// let issue: Issue = error.into();
```

does this not work? 🤔

Comment (KennedyTedesco) [reply src/issue.rs]: Sadly, not. But this does: `error.to_string().into()`.

Rust doesn't accept these implementations together:

```rust
impl<C: Into<String>> From<C> for Issue {
    fn from(message: C) -> Self {
        Issue::new(IssueSeverity::Error, message)
    }
}

impl<E: std::error::Error> From<E> for Issue {
    fn from(error: E) -> Self {
        Issue::new(IssueSeverity::Error, error.to_string())
    }
}
``` 

The error:

```
error[E0119]: conflicting implementations of trait `std::convert::From<_>` for type `issue::Issue`
   --> src/issue.rs:225:1
    |
219 | impl<C: Into<String>> From<C> for Issue {
    | --------------------------------------- first implementation here
...
225 | impl<E: std::error::Error> From<E> for Issue {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `issue::Issue`
``` 

Only accepts one or another.

Comment (KennedyTedesco) [reply src/issue.rs]: I think he does not know which implementation to use. Even this doesn't work:

```rust
impl From<String> for Issue {
    fn from(message: String) -> Self {
        Issue::new(IssueSeverity::Error, message)
    }
}

impl<E: std::error::Error> From<E> for Issue {
    fn from(error: E) -> Self {
        Issue::new(IssueSeverity::Error, error.to_string())
    }
}
```

Comment (KennedyTedesco) [reply src/issue.rs]: We can keep:

```rust
impl<E: std::error::Error> From<E> for Issue {
    fn from(error: E) -> Self {
        Issue::new(IssueSeverity::Error, error.to_string())
    }
}
```

And we can create a method on `Issue` called: `Issue::from_string("some issue")` that returns an issue of type error.

Or do you have another idea?

Comment (azjezz) [reply src/issue.rs]: yep, let's go with the last one

**Review by azjezz — APPROVED** [review]

**Pull request merged**